account_financial_report/report/general_ledger.py

```diff
@@ -838,6 +839,59 @@ def _get_report_values(self, docids, data):
                         account[grouped_by] = False
                         del account["list_grouped"]
         general_ledger = sorted(general_ledger, key=lambda k: k["code"])
+        # Set the bal_curr of the initial balance to 0 if it does not correspond
+        # (reducing the corresponding of the bal_curr of the initial balance).
+        for gl_item in general_ledger:
+            if (
+                not gl_item["currency_id"]
+                or gl_item["currency_id"] != company.currency_id
+            ):
+                gl_item["fin_bal"]["bal_curr"] -= gl_item["init_bal"]["bal_curr"]
```

Comment:
This cause an error if the check "Show foreign currency" is not enabled because 'gl_item["init_bal"]["bal_curr"]' will never be defined.

Reply:
Thank you, fixed in 15.0 (https://github.com/OCA/account-financial-reporting/pull/1256) and applied here.